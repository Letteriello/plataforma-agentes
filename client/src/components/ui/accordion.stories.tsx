import type { Meta, StoryObj } from '@storybook/react';
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from './accordion';

/**
 * A vertically stacked set of interactive headings that each reveal a section of content.
 * This component is built on top of Radix UI's Accordion primitive for accessibility.
 */
const meta: Meta<typeof Accordion> = {
  title: 'UI/Accordion',
  component: Accordion,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
    docs: {
      description: {
        component: `A vertically stacked set of interactive headings that each reveal a section of content. Built using Radix UI for accessibility.`,
      },
    },
  },
  argTypes: {
    type: {
      control: { type: 'radio' },
      options: ['single', 'multiple'],
      description: 'Determines whether one or multiple items can be opened at the same time.',
      table: {
        defaultValue: { summary: 'single' },
      },
    },
    collapsible: {
      control: 'boolean',
      description: 'When type is "single", allows closing the currently open item.',
      table: {
        defaultValue: { summary: 'false' },
      },
    },
  },
  decorators: [
    (Story) => (
      <div className="w-96">
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Accordion>;

const faqItems = [
  {
    value: 'item-1',
    question: 'Is it accessible?',
    answer: 'Yes. It adheres to the WAI-ARIA design pattern.',
  },
  {
    value: 'item-2',
    question: 'Is it styled?',
    answer: 'Yes. It comes with default styles that match other components.',
  },
  {
    value: 'item-3',
    question: 'Is it animated?',
    answer: 'Yes. It\'s animated by default, but this can be customized.',
  },
];

/**
 * The default behavior of the accordion allows only one item to be open at a time.
 * Clicking a new item will close the previous one.
 */
export const Single: Story = {
  args: {
    type: 'single',
    collapsible: true,
    className: 'w-full',
  },
  render: (args) => (
    <Accordion {...args}>
      {faqItems.map((item) => (
        <AccordionItem key={item.value} value={item.value}>
          <AccordionTrigger>{item.question}</AccordionTrigger>
          <AccordionContent>{item.answer}</AccordionContent>
        </AccordionItem>
      ))}
    </Accordion>
  ),
};

/**
 * By setting the `type` prop to `multiple`, you can allow multiple items to be
 * open simultaneously. The `collapsible` prop is ignored in this mode.
 */
export const Multiple: Story = {
  args: {
    type: 'multiple',
    className: 'w-full',
  },
  render: (args) => (
    <Accordion {...args}>
      {faqItems.map((item) => (
        <AccordionItem key={item.value} value={item.value}>
          <AccordionTrigger>{item.question}</AccordionTrigger>
          <AccordionContent>{item.answer}</AccordionContent>
        </A
