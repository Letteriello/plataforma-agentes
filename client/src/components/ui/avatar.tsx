import * as React from 'react'

import { cn } from '@/lib/utils'

// Componente Avatar simplificado que não depende do Radix UI
const Avatar = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement> & {
    // Props específicas do Avatar
    size?: 'sm' | 'md' | 'lg'
  }
>(({ className, size = 'md', ...props }, ref) => {
  const sizeClasses = {
    sm: 'h-8 w-8',
    md: 'h-10 w-10',
    lg: 'h-12 w-12',
  }

  return (
    <div
      ref={ref}
      className={cn(
        'relative flex shrink-0 overflow-hidden rounded-full bg-muted',
        sizeClasses[size],
        className,
      )}
      {...props}
    />
  )
})
Avatar.displayName = 'Avatar'

// Componente para imagem do avatar
const AvatarImage = React.forwardRef<
  HTMLImageElement,
  React.ImgHTMLAttributes<HTMLImageElement>
>(({ className, alt = '', ...props }, ref) => (
  <img
    ref={ref}
    className={cn('aspect-square h-full w-full object-cover', className)}
    alt={alt}
    {...props}
  />
))
AvatarImage.displayName = 'AvatarImage'

// Componente para fallback (quando a imagem não carrega)
const AvatarFallback = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      'flex h-full w-full items-center justify-center rounded-full bg-muted text-muted-foreground',
      className,
    )}
    {...props}
  />
))
AvatarFallback.displayName = 'AvatarFallback'

export { Avatar, AvatarFallback,AvatarImage }
