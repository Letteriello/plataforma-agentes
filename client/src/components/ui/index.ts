// UI Components
export * from "./button"
export * from "./card"
export * from "./input"
export * from "./textarea"
export * from "./select"
export * from "./badge"
export * from "./avatar"
export * from "./tooltip"

// New UI Components
export * from "./dialog"
export * from "./tabs"
export * from "./table"
export * from "./separator"

// Toaster and Toast components
export * from "./toaster"
export * from "./use-toast"

// Accordion components
export * from "./accordion";

// Form controls
export * from "./checkbox"

// Hooks and Utilities
export * from "@/lib/utils"
