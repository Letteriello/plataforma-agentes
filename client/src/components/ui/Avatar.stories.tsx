import type { Meta, StoryObj } from '@storybook/react';
import { Avatar, AvatarFallback, AvatarImage } from './avatar'; // Import all parts

const meta = {
  title: 'UI/Avatar',
  component: Avatar,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    size: {
      control: 'select',
      options: ['sm', 'md', 'lg'],
    },
    // Children for Avatar are typically AvatarImage and AvatarFallback
  },
  args: {
    size: 'md',
  },
} satisfies Meta<typeof Avatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: (args) => (
    <Avatar {...args}>
      <AvatarImage src="https://github.com/shadcn.png" alt="@shadcn" />
      <AvatarFallback>CN</AvatarFallback>
    </Avatar>
  ),
};

export const Small: Story = {
  render: (args) => (
    <Avatar {...args}>
      <AvatarImage src="https://github.com/storybookjs.png" alt="Storybook" />
      <AvatarFallback>SB</AvatarFallback>
    </Avatar>
  ),
  args: {
    size: 'sm',
  },
};

export const Large: Story = {
  render: (args) => (
    <Avatar {...args}>
      <AvatarImage src="https://github.com/vercel.png" alt="Vercel" />
      <AvatarFallback>VC</AvatarFallback>
    </Avatar>
  ),
  args: {
    size: 'lg',
  },
};

export const WithFallbackOnly: Story = {
  render: (args) => (
    <Avatar {...args}>
      {/* No AvatarImage provided, or if src was invalid */}
      <AvatarFallback>JD</AvatarFallback>
    </Avatar>
  ),
  args: {
    // You can also test with an invalid image src to see fallback
    // src: "https://invalid-url-to-force-fallback.png"
  },
};

export const FallbackWithLongerText: Story = {
  render: (args) => (
    <Avatar {...args}>
      <AvatarFallback>USER</AvatarFallback>
    </Avatar>
  ),
  args: {
    size: 'lg',
  },
};

export const FallbackWithDelay: Story = {
  render: (args) => (
    <Avatar {...args}>
      <AvatarImage src="https://via.placeholder.com/150/FFFFFF/000000?text=Loading...&Pdelay=2000" alt="Delayed Load" />
      <AvatarFallback>LD</AvatarFallback>
    </Avatar>
  ),
  args: {
    // Note: The Pdelay parameter in placeholder.com URLs simulates network delay.
    // Storybook might not show the fallback state dynamically unless the image actually fails or is slow.
    // To reliably show fallback, omit AvatarImage or provide an invalid src.
  },
  parameters: {
    docs: {
      description: {
        story: 'This story attempts to show a fallback during a slow image load. ' +
                 'However, reliably demonstrating this in Storybook might require an actual failing or very slow image URL, ' +
                 'or manipulating the AvatarImage component to simulate a load failure.',
      },
    },
  },
};
