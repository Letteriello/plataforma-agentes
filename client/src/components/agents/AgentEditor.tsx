import { useState, useEffect } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { useToast } from '@/components/ui/use-toast';
import { Button } from '@/components/ui/button';
import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs';
import { useFormContext, FormProvider } from 'react-hook-form'; // Added FormProvider
import { Input } from '@/components/ui/input';
import { Textarea } from '@/components/ui/textarea';
import { FormField, FormItem, FormLabel, FormControl, FormMessage, FormDescription } from '@/components/ui/form';
import { Card, CardContent, CardHeader, CardTitle, CardDescription } from '@/components/ui/card';

import { LLMAgentForm } from './forms/LLMAgentForm';
import { BaseAgentForm } from './forms/BaseAgentForm'; // This should provide useForm methods
import { LLMAgent, createDefaultAgent, LLMAgentSchema } from '@/types/agents';
import AgentDeployTab from './AgentDeployTab';

const WIZARD_STEPS = ['identidade', 'instrucoes', 'ferramentas', 'revisao', 'deploy'];
const STEP_LABELS: { [key: string]: string } = {
  identidade: 'Identidade',
  instrucoes: 'Instruções',
  ferramentas: 'Ferramentas',
  revisao: 'Revisão',
  deploy: 'Deploy',
};

// Mock data - replace with actual API calls
const mockAgents: LLMAgent[] = [
  {
    id: '1',
    name: 'Customer Support Bot',
    description: 'Handles customer inquiries and support tickets',
    type: 'llm',
    model: 'gemini-1.5-pro',
    temperature: 0.7,
    maxTokens: 1024,
    topP: 1,
    topK: 40,
    stopSequences: [],
    frequencyPenalty: 0,
    presencePenalty: 0,
    instruction: 'You are a helpful customer support assistant. Be polite and professional.',
    systemPrompt: '## Guidelines\n- Always verify customer information\n- Escalate complex issues to a human agent\n- Provide clear next steps',
    version: '1.0.0',
    isPublic: false,
    tags: ['support', 'customer-service'],
    createdAt: '2024-01-01T00:00:00Z',
    updatedAt: '2024-01-01T00:00:00Z',
  },
];

type AgentEditorProps = {
  mode?: 'create' | 'edit';
};

export function AgentEditor({ mode = 'create' }: AgentEditorProps) {
  const { id } = useParams<{ id: string }>();
  const navigate = useNavigate();
  const { toast } = useToast();
  const [isLoading, setIsLoading] = useState(true);
  const [agent, setAgent] = useState<LLMAgent | null>(null);
  const [activeTab, setActiveTab] = useState(WIZARD_STEPS[0]);

  // Load agent data in edit mode
  useEffect(() => {
    if (mode === 'edit' && id) {
      // In a real app, this would be an API call
      const foundAgent = mockAgents.find(a => a.id === id) || null;
      setAgent(foundAgent);
      setIsLoading(false);
    } else if (mode === 'create') {
      setAgent(createDefaultAgent('llm'));
      setIsLoading(false);
    }
  }, [id, mode]);

  const currentStepIndex = WIZARD_STEPS.indexOf(activeTab);

  const handleNextStep = () => {
    if (currentStepIndex < WIZARD_STEPS.length - 1) {
      setActiveTab(WIZARD_STEPS[currentStepIndex + 1]);
    }
  };

  const handlePreviousStep = () => {
    if (currentStepIndex > 0) {
      setActiveTab(WIZARD_STEPS[currentStepIndex - 1]);
    }
  };

  const handleSubmit = async (values: LLMAgent) => {
    try {
      setIsLoading(true);
      // In a real app, this would be an API call
      console.log('Saving agent:', values);
      
      // Simulate API call
      await new Promise(resolve => setTimeout(resolve, 1000));
      
      toast({
        title: 'Sucesso',
        description: `Agente ${mode === 'create' ? 'criado' : 'atualizado'} com sucesso`,
      });
      
      // Navigate back to agents list or to the new agent's edit page
      navigate('/agents');
    } catch (error) {
      console.error('Error saving agent:', error);
      toast({
        title: 'Erro',
        description: `Falha ao ${mode === 'create' ? 'criar' : 'atualizar'} o agente. Por favor, tente novamente.`,
        variant: 'destructive',
      });
    } finally {
      setIsLoading(false);
    }
  };

  const handleCancel = () => {
    navigate('/agents');
  };

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (!agent && mode === 'edit') {
    return <div>Agent not found</div>;
  }

  // Component to display review, needs access to form context
  const ReviewStep = () => {
    const { watch } = useFormContext<LLMAgent>();
    const watchedAgentData = watch(); // Watch all fields

    return (
      <Card>
        <CardHeader>
          <CardTitle>Revise Seu Agente</CardTitle>
          <CardDescription>Por favor, revise a configuração do agente antes de prosseguir.</CardDescription>
        </CardHeader>
        <CardContent className="space-y-3">
          <div>
            <h4 className="font-semibold">Nome:</h4>
            <p>{watchedAgentData.name || 'Não definido'}</p>
          </div>
          <div>
            <h4 className="font-semibold">Descrição:</h4>
            <p>{watchedAgentData.description || 'Não definido'}</p>
          </div>
          <div>
            <h4 className="font-semibold">URL do Avatar:</h4>
            <p>{watchedAgentData.avatarUrl || 'Não definido'}</p>
          </div>
          <div>
            <h4 className="font-semibold">Instruções:</h4>
            <pre className="whitespace-pre-wrap p-2 bg-gray-100 dark:bg-gray-900 rounded-md">
              {watchedAgentData.instruction || 'Não definido'}
            </pre>
          </div>
          {watchedAgentData.systemPrompt && (
            <div>
              <h4 className="font-semibold">Prompt de Sistema:</h4>
              <pre className="whitespace-pre-wrap p-2 bg-gray-100 dark:bg-gray-900 rounded-md">
                {watchedAgentData.systemPrompt}
              </pre>
            </div>
          )}
        </CardContent>
      </Card>
    );
  };

  return (
    <div className="container mx-auto py-8">
      <div className="flex justify-between items-center mb-6">
        <h1 className="text-2xl font-bold">
          {mode === 'create' ? 'Criar Novo Agente' : `Editar Agente: ${agent?.name}`}
        </h1>
      </div>

      <BaseAgentForm
        id="agent-form"
        defaultValues={agent}
        onSubmit={handleSubmit}
        schema={LLMAgentSchema}
        className="space-y-6"
      >
        <Tabs
          value={activeTab}
          onValueChange={setActiveTab}
          // className="space-y-6" // Moved to BaseAgentForm or manage spacing internally
        >
          <TabsList>
            {WIZARD_STEPS.map(step => (
              <TabsTrigger key={step} value={step}>
                {STEP_LABELS[step]}
              </TabsTrigger>
            ))}
          </TabsList>

          <TabsContent value="identidade">
            <Card>
              <CardHeader>
                <CardTitle>Identidade do Agente</CardTitle>
                <CardDescription>Defina o nome, a descrição e o avatar do seu agente.</CardDescription>
              </CardHeader>
              <CardContent className="space-y-4">
                <FormField
                  name="name"
                  render={({ field, formState }) => ( // control is implicitly from BaseAgentForm's FormProvider
                    <FormItem>
                      <FormLabel>Nome do Agente</FormLabel>
                      <FormControl>
                        <Input placeholder="Meu Agente Incrível" {...field} />
                      </FormControl>
                      <FormMessage>{formState.errors.name?.message?.toString()}</FormMessage>
                    </FormItem>
                  )}
                />
                <FormField
                  name="description"
                  render={({ field, formState }) => (
                    <FormItem>
                      <FormLabel>Descrição</FormLabel>
                      <FormControl>
                        <Textarea placeholder="Descreva o que seu agente faz." {...field} rows={3} />
                      </FormControl>
                      <FormMessage>{formState.errors.description?.message?.toString()}</FormMessage>
                    </FormItem>
                  )}
                />
                <FormField
                  name="avatarUrl"
                  render={({ field, formState }) => (
                    <FormItem>
                      <FormLabel>URL do Avatar (Opcional)</FormLabel>
                      <FormControl>
                        <Input placeholder="https://example.com/avatar.png" {...field} value={field.value ?? ''} />
                      </FormControl>
                      <FormMessage>{formState.errors.avatarUrl?.message?.toString()}</FormMessage>
                    </FormItem>
                  )}
                />
              </CardContent>
            </Card>
          </TabsContent>

          <TabsContent value="instrucoes">
            <Card>
              <CardHeader>
                <CardTitle>Instruções do Agente</CardTitle>
                <CardDescription>Forneça as instruções e o prompt do sistema para guiar o comportamento do agente.</CardDescription>
              </CardHeader>
              <CardContent className="space-y-4">
                <FormField
                  name="instruction"
                  render={({ field, formState }) => (
                    <FormItem>
                      <FormLabel>Instruções</FormLabel>
                      <FormControl>
                        <Textarea
                          placeholder="Você é um assistente útil que..."
                          {...field}
                          rows={6}
                          className="font-mono text-sm"
                        />
                      </FormControl>
                      <FormDescription>Instruções de sistema que guiam o comportamento do agente.</FormDescription>
                      <FormMessage>{formState.errors.instruction?.message?.toString()}</FormMessage>
                    </FormItem>
                  )}
                />
                <FormField
                  name="systemPrompt"
                  render={({ field, formState }) => (
                    <FormItem>
                      <FormLabel>Prompt de Sistema (Opcional)</FormLabel>
                      <FormControl>
                        <Textarea
                          placeholder="## Contexto\n- ...\n## Regras\n- ..."
                          {...field}
                          rows={6}
                          className="font-mono text-sm"
                        />
                      </FormControl>
                      <FormDescription>Instruções adicionais de nível de sistema, ex: em Markdown.</FormDescription>
                      <FormMessage>{formState.errors.systemPrompt?.message?.toString()}</FormMessage>
                    </FormItem>
                  )}
                />
              </CardContent>
            </Card>
          </TabsContent>

          <TabsContent value="ferramentas">
            <Card>
              <CardHeader><CardTitle>Ferramentas</CardTitle></CardHeader>
              <CardContent>
                <p className="text-muted-foreground">Em breve: Configure ferramentas e capacidades para o seu agente.</p>
              </CardContent>
            </Card>
          </TabsContent>

          <TabsContent value="revisao">
            <ReviewStep />
          </TabsContent>

          <TabsContent value="deploy">
            <AgentDeployTab />
          </TabsContent>
        </Tabs>

        {/* LLMAgentForm for model settings, hidden from wizard steps but part of the form */}
        {/* This ensures these fields are submitted. We might move them into a wizard step later. */}
        {/* For now, they are not directly part of the wizard UI but are part of the form */}
        <div className={activeTab === 'identidade' || activeTab === 'instrucoes' ? 'block mt-6' : 'hidden'}>
           {/* Show Model config only on first two steps or a dedicated step later */}
          <LLMAgentForm />
        </div>

        {/* Navigation buttons are now OUTSIDE Tabs but INSIDE BaseAgentForm */}
        <div className="flex justify-between items-center mt-8 pt-6 border-t border-gray-200 dark:border-gray-700">
          <div>
            <Button
              variant="outline"
              onClick={handlePreviousStep}
              disabled={currentStepIndex === 0 || isLoading}
              className={currentStepIndex === 0 ? 'invisible' : ''}
            >
              Anterior
            </Button>
          </div>
          <div className="space-x-3">
            <Button variant="outline" onClick={handleCancel} disabled={isLoading}>
              Cancelar
            </Button>
            {currentStepIndex === WIZARD_STEPS.length - 1 ? (
              <Button form="agent-form" type="submit" disabled={isLoading}>
                {isLoading ? (mode === 'create' ? 'Criando...' : 'Salvando...') : (mode === 'create' ? 'Criar Agente' : 'Salvar Alterações')}
              </Button>
            ) : (
              <Button type="button" onClick={handleNextStep} disabled={currentStepIndex === WIZARD_STEPS.length - 1 || isLoading}>
                Próximo
              </Button>
            )}
          </div>
        </div>
      </BaseAgentForm>
    </div>
  );
}
