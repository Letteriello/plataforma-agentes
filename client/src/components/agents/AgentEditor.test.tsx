import React from 'react'
import { render, act } from '@testing-library/react'
import { vi } from 'vitest'
import { MemoryRouter } from 'react-router-dom'
import { AgentEditor } from './AgentEditor'
import { Toaster } from '@/components/ui/toaster' // Needed if toasts are rendered

// Mock dependencies
vi.mock('@/components/ui/use-toast', () => ({
  useToast: () => ({
    toast: vi.fn(),
    toasts: [], // Add the missing toasts array
  }),
}))

vi.mock('react-router-dom', async (importOriginal) => {
  const actual = await importOriginal()
  return {
    // @ts-ignore
    ...actual, // import and retain default behavior
    useNavigate: () => vi.fn(),
    useParams: () => ({ id: undefined }), // Mock for 'create' mode
  }
})

// Mock a sub-component that might be complex or not relevant to this snapshot
vi.mock('@/components/agents/forms/LLMAgentForm', () => ({
  LLMAgentForm: () => (
    <div data-testid="mock-llm-agent-form">LLMAgentForm Mock</div>
  ),
}))

// Mock another sub-component
vi.mock('@/components/agents/AgentDeployTab', () => ({
  __esModule: true, // This is important for modules with default exports
  default: () => (
    <div data-testid="mock-agent-deploy-tab">AgentDeployTab Mock</div>
  ),
}))

describe('AgentEditor', () => {
  it('should match snapshot for create mode', async () => {
    let container
    // Use act for updates related to useEffect and state changes
    // The AgentEditor initializes a default agent which might involve state updates
    await act(async () => {
      const rendered = render(
        <MemoryRouter>
          <AgentEditor />
          <Toaster />{' '}
          {/* Include Toaster if toast messages can be triggered on render */}
        </MemoryRouter>,
      )
      container = rendered.container
    })
    expect(container).toMatchSnapshot()
  })
})
