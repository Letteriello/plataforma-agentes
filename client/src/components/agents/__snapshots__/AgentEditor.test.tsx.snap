// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentEditor > should match snapshot for create mode 1`] = `
<div>
  <div
    class="container mx-auto py-8"
  >
    <div
      class="flex justify-between items-center mb-6"
    >
      <h1
        class="text-2xl font-bold"
      >
        Criar Novo Agente
      </h1>
    </div>
    <form
      class="space-y-6 space-y-6"
    >
      <div
        class="space-y-4"
      >
        <div
          data-orientation="horizontal"
          dir="ltr"
        >
          <div
            aria-orientation="horizontal"
            class="inline-flex h-10 items-center justify-center rounded-md bg-muted p-1 text-muted-foreground"
            data-orientation="horizontal"
            role="tablist"
            style="outline: none;"
            tabindex="0"
          >
            <button
              aria-controls="radix-:r0:-content-identidade"
              aria-selected="true"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-state="active"
              id="radix-:r0:-trigger-identidade"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Identidade
            </button>
            <button
              aria-controls="radix-:r0:-content-instrucoes"
              aria-selected="false"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-state="inactive"
              id="radix-:r0:-trigger-instrucoes"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Instruções
            </button>
            <button
              aria-controls="radix-:r0:-content-ferramentas"
              aria-selected="false"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-state="inactive"
              id="radix-:r0:-trigger-ferramentas"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Ferramentas
            </button>
            <button
              aria-controls="radix-:r0:-content-revisao"
              aria-selected="false"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-state="inactive"
              id="radix-:r0:-trigger-revisao"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Revisão
            </button>
            <button
              aria-controls="radix-:r0:-content-deploy"
              aria-selected="false"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-state="inactive"
              id="radix-:r0:-trigger-deploy"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Deploy
            </button>
          </div>
          <div
            aria-labelledby="radix-:r0:-trigger-identidade"
            class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
            data-orientation="horizontal"
            data-state="active"
            id="radix-:r0:-content-identidade"
            role="tabpanel"
            style="animation-duration: 0s;"
            tabindex="0"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Identidade do Agente
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Defina o nome, a descrição e o avatar do seu agente.
                </p>
              </div>
              <div
                class="p-6 pt-0 space-y-4"
              >
                <div
                  class="space-y-2"
                >
                  <label
                    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    for=":r6:-form-item"
                  >
                    Nome do Agente
                  </label>
                  <input
                    aria-describedby=":r6:-form-item-description"
                    aria-invalid="false"
                    class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                    id=":r6:-form-item"
                    name="name"
                    placeholder="Meu Agente Incrível"
                    value="New Llm Agent"
                  />
                </div>
                <div
                  class="space-y-2"
                >
                  <label
                    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    for=":r7:-form-item"
                  >
                    Descrição
                  </label>
                  <textarea
                    aria-describedby=":r7:-form-item-description"
                    aria-invalid="false"
                    class="flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                    id=":r7:-form-item"
                    name="description"
                    placeholder="Descreva o que seu agente faz."
                    rows="3"
                  />
                </div>
                <div
                  class="space-y-2"
                >
                  <label
                    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                    for=":r8:-form-item"
                  >
                    URL do Avatar (Opcional)
                  </label>
                  <input
                    aria-describedby=":r8:-form-item-description"
                    aria-invalid="false"
                    class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                    id=":r8:-form-item"
                    name="avatarUrl"
                    placeholder="https://example.com/avatar.png"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="radix-:r0:-trigger-instrucoes"
            class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
            data-orientation="horizontal"
            data-state="inactive"
            hidden=""
            id="radix-:r0:-content-instrucoes"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-labelledby="radix-:r0:-trigger-ferramentas"
            class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
            data-orientation="horizontal"
            data-state="inactive"
            hidden=""
            id="radix-:r0:-content-ferramentas"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-labelledby="radix-:r0:-trigger-revisao"
            class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
            data-orientation="horizontal"
            data-state="inactive"
            hidden=""
            id="radix-:r0:-content-revisao"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-labelledby="radix-:r0:-trigger-deploy"
            class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
            data-orientation="horizontal"
            data-state="inactive"
            hidden=""
            id="radix-:r0:-content-deploy"
            role="tabpanel"
            tabindex="0"
          />
        </div>
        <div
          class="block mt-6"
        >
          <div
            data-testid="mock-llm-agent-form"
          >
            LLMAgentForm Mock
          </div>
        </div>
        <div
          class="flex justify-between items-center mt-8 pt-6 border-t border-gray-200 dark:border-gray-700"
        >
          <div>
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background border border-input hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4 invisible"
              disabled=""
            >
              Anterior
            </button>
          </div>
          <div
            class="space-x-3"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background border border-input hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4"
            >
              Cancelar
            </button>
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
              type="button"
            >
              Próximo
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex justify-end space-x-3 pt-4"
      >
        <button
          class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background border border-input hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4"
          type="button"
        >
          Cancel
        </button>
        <button
          class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>

</div>
`;
